### DISCUSSION TITLE
best_f vs current_value

### DISCUSSION THREAD

**Discussion by samuelkim16** [meta-pytorch/botorch#2513] [Q&A] [answered]
What is the difference in `best_f` used in acquisition functions such as `qExpectedImprovement`, and `current_value` used in acquisition functions such as `qKnowledgeGradient`?

**Comment by Balandat** [comment, marked as answer]
They're essentially the same in spirit / semantically. For analytic EI, `best_f` is indeed the best observed function value so far of the black box function `f`. But for the MC acquisition functions such as qEI and also qKG, we allow to pass in some generic `objective` function, so it's the best objective rather than the best function value in general. Arguably, one should probably rename the arg to qEI for consistency, but as it's the generalization of EI specifically this is done for historical reasons.

**Reply by samuelkim16** [reply]
Thank you! Is there a reason why this tutorial on [Multi-Fidelity BO](https://botorch.org/tutorials/discrete_multi_fidelity_bo) re-calculates the `current_value` in each iteration despite a separate `objective` function not being used (as far as I can tell)? Could one just keep track of the `current_value`?

**Reply by Balandat** [reply]
I am not sure I understand what you mean by "Could one just keep track of the current_value".

The `current_value` depends on all of the data observed so far, so you need to re-calculate it whenever you get new data. Note that there is an additional complication here with multi-fidelity methods in that we can't just optimize the posterior mean of the model across all parameters, but instead need to fix the target fidelity and optimize w.r.t. the other parameters - this is why `FixedFeatureAcquisitionFunction` is used here.

**Reply by samuelkim16** [reply]
Ah I think I misunderstood `current value`. This is the _expected_ best function value, as opposed to `best_f` which is the _observed_ best function value - is that right? Which is why it needs to be re-calculated in the case of knowledge gradient.
